packages/editor-ui/src/router.ts

```diff
@@ -293,6 +294,28 @@ export const routes = [
 			},
 		],
 	},
+	{
+		path: '/workflow/:name/history/:id?',
+		name: VIEWS.WORKFLOW_HISTORY,
+		components: {
+			default: WorkflowHistory,
+			sidebar: MainSidebar,
+		},
+		meta: {
+			keepWorkflowAlive: true,
+			permissions: {
+				allow: {
+					loginStatus: [LOGIN_STATUS.LoggedIn],
+				},
+				deny: {
+					shouldDeny: () =>
+						!useSettingsStore().isEnterpriseFeatureEnabled(
+							EnterpriseEditionFeature.WorkflowHistory,
+						),
+				},
```

Comment:
Would it be clearer to invert this and add it as `shouldAllow` under `allow`?

Reply:
Just copying previous routes, I'm not sure if what you say behaves the same way or how you think but I agree with you

Reply:
Just out of curiosity I checked how it works and found [this](https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/utils/userUtils.ts#L138)

If I move my condition to `allow.shouldAllow` I should move `allow.loginStatus` to 'deny.loginStatus' and check for `LOGIN_STATUS.LoggedOut`

Let me try

Reply:
Apart of it's not working (app fails with some errors I don't yet want to fix) the problem here is `allow` conditions are evaluated in an `OR` relation so setting `loginStatus` and `shouldAllow` rule would allow just logged in users to see this view.
Leaving it like this for now

Reply:
Actually, for a feature to be enabled for a user, they must be logged in, I guess

Reply:
Still not working, if I just set `allow` and don't have `deny` (maybe not related) the app crashes so leaving it like this for now

Reply:
Thanks for trying. What an odd API that we cannot simply express two cumulative conditions to allow.